src/services/paras/ParasService.spec.ts

```diff
@@ -0,0 +1,241 @@
+			const emptyLeasesAt = () => Promise.resolve().then(() => []);
+
+			(mockApi.query.slots.leases.at as unknown) = emptyLeasesAt;
+
+			const expectedResponse = {
+				at: expectedAt,
+				leases: null,
+				paraLifeCycle: 'Onboarding',
+				onboardingAs: 'parachain',
+			};
+
+			const response = await parasService.leaseInfo(blockHash789629, paraId);
+
+			expect(sanitizeNumbers(response)).toMatchObject(expectedResponse);
+
+			(mockApi.query.slots.leases.at as unknown) = slotsLeasesAt;
+		});
+	});
+
+	describe('ParasService.leasesCurrent', () => {
+		it('Should return the correct entries for leasesCurrent', async () => {
+			const expectedResponse = {
+				at: expectedAt,
+				leasePeriodIndex: '39',
+				endOfLeasePeriod: '800000',
+				currentLeaseHolders: ['199', '200'],
+			};
+
+			const response = await parasService.leasesCurrent(blockHash789629, true);
+
+			expect(sanitizeNumbers(response)).toMatchObject(expectedResponse);
+		});
+
+		it('Should return the correct response exlcuding currentLeaseHolders', async () => {
+			const expectedResponse = {
+				at: expectedAt,
+				leasePeriodIndex: '39',
+				endOfLeasePeriod: '800000',
+				currentLeaseHolders: undefined,
+			};
+
+			const response = await parasService.leasesCurrent(blockHash789629, false);
+
+			expect(sanitizeNumbers(response)).toMatchObject(expectedResponse);
+		});
+	});
+
+	describe('ParasService.paras', () => {
+		it('Should return correct ParaLifecycles response', async () => {
+			const expectedResponse = {
+				at: expectedAt,
+				paras: [
+					{
+						paraId: '199',
+						paraLifeCycle: 'Onboarding',
+						onboardingAs: 'parachain',
+					},
+					{
+						paraId: '200',
+						paraLifeCycle: 'Parathread',
+						onboardingAs: undefined,
+					},
+				],
+			};
+
+			const response = await parasService.paras(blockHash789629);
+
+			expect(sanitizeNumbers(response)).toMatchObject(expectedResponse);
+		});
+	});
+
+	describe('ParasService.auctionsCurrent', () => {
+		it('Should return to correct data during an ongoing auction', async () => {
+			const expectedResponse = {
+				at: expectedAt,
+				beginEnd: '39',
+				finishEnd: '20039',
+				phase: 'ending',
+				auctionIndex: '4',
+				leasePeriods: ['1000', '1001', '1002', '1003'],
+				winning: [
+					{
+						bid: {
+							accountId: '5CXFhuwT7A1ge4hCa23uCmZWQUebEZSrFdBEE24C41wmAF4N',
+							amount: '1000000',
+							paraId: '199',
+						},
+						leaseSet: ['1000'],
+					},
+					{
+						bid: {
+							accountId: '5ESEa1HV8hyG6RTXgwWNUhu5fXvkHBfEJKjw3hKmde7fXdHQ',
+							amount: '2000000',
+							paraId: '200',
+						},
+						leaseSet: ['1000', '1001'],
+					},
+				],
+			};
+
+			const response = await parasService.auctionsCurrent(blockHash789629);
+
+			expect(sanitizeNumbers(response)).toMatchObject(expectedResponse);
+		});
+
+		it('Should return the correct null values when auctionInfo is not of type Option', async () => {
```

Comment:
```suggestion
		it('Should return the correct null values when auctionInfo is `None`', async () => {
```
It should be of type `Option`: `Option::None`. In JSON though that gets converted into `null`.

Reply:
The `sanitizeNumber` specs are exhaustive for this stuff. See

https://github.com/paritytech/substrate-api-sidecar/blob/ae74ef77cf05c68fc7a552b8cccc200262f721be/src/sanitize/sanitizeNumbers.spec.ts#L572-L575

Reply:
ahh gotcha, yea that makes sense. Empty Options are returned as null with sanitize numbers :)!